File: backend/src/colyseus/schemas/PlayerSchema.ts
import { Schema, type } from '@colyseus/schema'

export class PlayerSchema extends Schema {
  @type('string') playerId: string = ''
  @type('string') username: string = ''
  @type('number') score: number = 0
  @type('boolean') isBot: boolean = false
  @type('boolean') isConnected: boolean = true
  @type('number') slotNumber: number = 0
  @type('string') teamColor: string = 'red'
}

File: game/Assets/_Game/Scripts/Network/Schema/PlayerSchema.cs
// Generated schema — mirrors backend/src/colyseus/schemas/PlayerSchema.ts

using Colyseus.Schema;
#if UNITY_5_3_OR_NEWER
using UnityEngine.Scripting;
#endif

public partial class PlayerSchema : Schema
{
#if UNITY_5_3_OR_NEWER
    [Preserve]
#endif
    public PlayerSchema() { }

    [Type(0, "string")]  public string playerId    = default;
    [Type(1, "string")]  public string username    = default;
    [Type(2, "number")]  public float  score       = 0;
    [Type(3, "boolean")] public bool   isBot       = false;
    [Type(4, "boolean")] public bool   isConnected = false;
    [Type(5, "number")]  public float  slotNumber  = 0;
    [Type(6, "string")]  public string teamColor   = default;
    [Type(7, "number")]  public float  baseX       = 0;
    [Type(8, "number")]  public float  baseY       = 0;
}
